sagemaker-python-sdk/mxnet_gluon_mnist/mxnet_mnist_with_gluon_1.3.ipynb

```diff
@@ -0,0 +1,1139 @@
+      "    :return: response payload and content type.\r\n",
+      "    \"\"\"\r\n",
+      "    # we can use content types to vary input/output handling, but\r\n",
+      "    # here we just assume json for both\r\n",
+      "    parsed = json.loads(data)\r\n",
+      "    nda = mx.nd.array(parsed)\r\n",
+      "    output = net(nda)\r\n",
+      "    prediction = mx.nd.argmax(output, axis=1)\r\n",
+      "    response_body = json.dumps(prediction.asnumpy().tolist()[0])\r\n",
+      "    return response_body, output_content_type\r\n",
+      "\r\n",
+      "\r\n",
+      "if __name__ == '__main__':\r\n",
+      "    # mxnet-1.3 requirements on https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/mxnet/README.rst\r\n",
+      "    parser = argparse.ArgumentParser()\r\n",
+      "\r\n",
+      "    # hyperparameters\r\n",
+      "    parser.add_argument('--sm-hps', type=json.loads, default=os.environ['SM_HPS'])\r\n",
+      "\r\n",
+      "    # input data and model directories\r\n",
+      "    parser.add_argument('--model-dir', type=str, default=os.environ['SM_MODEL_DIR'])\r\n",
+      "    parser.add_argument('--training', type=str, default=os.environ['SM_CHANNEL_TRAINING'])\r\n",
+      "\r\n",
+      "    # other stuff\r\n",
+      "    parser.add_argument('--sm-current-host', type=str, default=os.environ['SM_CURRENT_HOST'])\r\n",
+      "    parser.add_argument('--sm-hosts', type=json.loads, default=os.environ['SM_HOSTS'])\r\n",
+      "    parser.add_argument('--sm-num-gpus', type=int, default=os.environ['SM_NUM_GPUS'])\r\n",
+      "\r\n",
+      "    args, _ = parser.parse_known_args()\r\n",
+      "\r\n",
+      "    print(args)\r\n",
+      "\r\n",
+      "    channel_input_dirs = {'training': args.training}\r\n",
+      "\r\n",
+      "    train_ret = train(args.sm_current_host, channel_input_dirs, args.sm_hps, args.sm_hosts, args.sm_num_gpus)\r\n",
+      "    save(train_ret, args.model_dir)\r\n"
+     ]
+    }
+   ],
+   "source": [
+    "!cat 'mnist.py'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Run the training script on SageMaker\n",
+    "\n",
+    "The ```MXNet``` class allows us to run our training function on SageMaker local mode. We need to configure it with our training script, an IAM role, the number of training instances, and the training instance type.  This is the the only difference from [mnist_with_gluon.ipynb](./mnist_with_gluon.ipynb).  Instead of ``train_instance_type='ml.c4.xlarge'``, we set it to ``train_instance_type='local'``.  For local training with GPU, we could set this to \"local_gpu\".  In this case, `instance_type` was set above based on your whether you're running a GPU instance."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m = MXNet(\"mnist.py\",\n",
+    "          role=role,\n",
+    "          train_instance_count=1,\n",
```

Comment:
it's strange that above you say this needs to be changed to 1 for GPU when it's already 1 here. maybe the wording above can be updated